socketmode/socketmode_eventhandler.go

```diff
@@ -0,0 +1,218 @@
+package socketmode
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/slack-go/slack"
+	"github.com/slack-go/slack/slackevents"
+)
+
+type SocketmodeHandler struct {
+	Client *Client
+
+	EventMap                       map[EventType][]SocketmodeHandlerFunc
+	InteractionEventMap            map[slack.InteractionType][]SocketmodeHandlerFunc
+	InteractionBlockActionEventMap map[string][]SocketmodeHandlerFunc
+	EventApiMap                    map[slackevents.EventAPIType][]SocketmodeHandlerFunc
+	SlashCommandMap                map[string][]SocketmodeHandlerFunc
```

Comment:
[question]
Is it intended that it accepts multiple handlers for a single event, interaction, block action or slack command?

Reply:
Yes, as mentioned before I want to be as flexible as possible. Additionally, this is required to achieve separation of concerns. Controllers/Services are not aware of what other controllers registered.

Reply:
I think it is worth mentioning that wrote a more in-depth tutorial here: https://github.com/xNok/slack-go-demo-socketmode. It may better illustrate the intent. Please feel free to comment on that as well.

Reply:
Umm... 
For example,` action_id` is defined as unique, but if multiple handlers can be prepared for a single `action_id`, it may cause unintentional duplicate registrations and other disadvantages.

> Controllers/Services are not aware of what other controllers registered.

How about a way to return an error when registering duplicates, like the net/http package?

Reply:
https://github.com/golang/go/blob/master/src/net/http/server.go#L2429-L2455

Reply:
I think this is the way forward it makes sense to me.